### PR TITLE
Update documentation for triggering workspace runs with Git tags

### PR DESCRIPTION
**Pull request by hashimoon**
### What

Add documentation for how to trigger a Workspace run through a matching Git tag.
- Workspaces

### Why

A new feature was added to trigger runs through Git tags

### Screenshots

----------

### Merge Checklist
_If items do not apply to your changes, add (N/A) and mark them as complete._

#### Pull Request
- [x] One or more labels describe the type of change (e.g. clarification) and associated product (e.g. tfc).
- [x] Description links to related pull requests or issues, if any.

#### Content
- [x] Redirects have been added for moved, renamed, or deleted pages. This requires a separate PR in the [`terraform-website` repository](https://github.com/hashicorp/terraform-website) `redirects.next.js` file.
- [x] API documentation and the API Changelog have been updated. 
- [x] Links to related content where appropriate (e.g., API endpoints, permissions, etc.).
- [x] Pages with related content are updated and link to this content when appropriate.
- [x] Sidebar navigation files have been updated for added, deleted, reordered, or renamed pages.
- [x] New pages have metadata (page name and description) at the top.
- [x] New images are 2048 px wide. They have HashiCorp standard annotation color (#F92672) and format (rectangle with rounded corners), blurred sensitive details (e.g. credentials, usernames, user icons), and descriptive alt text in the markdown for accessibility.
- [x] New code blocks have the correct syntax and line breaks to eliminate horizontal scroll bars.
- [x] UI elements (button names, page names, etc.) are bolded.
- [x] The Vercel website preview successfully deployed.

#### Reviews
- [x] I or someone else reviewed the content for technical accuracy.
- [x] I or someone else reviewed the content for typos, punctuation, spelling, and grammar.

### CODE REVIEW

**Review by omarismail — APPROVED** [review]

**Review by paulanunda — APPROVED** [review]

File: website/docs/cloud-docs/api-docs/workspaces.mdx [hunk 1]
```diff
@@ -86,6 +86,7 @@ By supplying the necessary attributes under a `vcs-repository` object, you can c
...
 | `data.attributes.vcs-repo.identifier`         | string  |           | A reference to your VCS repository in the format :org/:repo where :org and :repo refer to the organization and repository in your VCS provider. The format for Azure DevOps is `:org/:project/_git/:repo`.                                                                                                                                                                                                                                                                                                      |
+| `data.attributes.vcs-repo.tags-regex`         | string  | (nothing) | A regular expression used to match Git tags. When this value is present and a VCS event is received that contains a matching Git tag for the regular expression, a Terraform Cloud run will be triggered.                                                                                                                                                                                                                                                                                                       |
```
Comment (mgarrell777) [inline website/docs/cloud-docs/api-docs/workspaces.mdx:89]: Hi @hashimoon 

I'm Matthew, new technical writer on the Terraform team. Overall, very good stuff here. I just have a couple of pretty small wording changes.

For the Description of `data.attributes.vcs-repo.tags-regex`, I'd suggest rewording to "Terraform Cloud triggers a run when this value is present and a VCS event occurs that contains a matching Git tag for the regular expression."

For the Description of `data.attributes.working-directory`, I'd suggest rewording to "A relative path from which Terraform executes. The path defaults to your repository's root and is typically set to a subdirectory matching the environment when multiple environments exist within the same repository."

Thanks very much for your time and effort.

File: website/docs/cloud-docs/api-docs/workspaces.mdx [hunk 2]
```diff
@@ -484,6 +512,7 @@ Properties without a default value are required.
...
 | `data.attributes.vcs-repo.identifier`         | string         | (previous value) | A reference to your VCS repository in the format :org/:repo where :org and :repo refer to the organization and repository in your VCS provider. The format for Azure DevOps is `:org/:project/_git/:repo`.                                                                                                                                                                                  |
+| `data.attributes.vcs-repo.tags-regex`         | string         | (previous value) | A regular expression used to match Git tags. When this value is present and a VCS event is received that contains a matching Git tag for the regular expression, a Terraform Cloud run will be triggered.                                                                                                                                                                                   |
```
Comment (mgarrell777) [inline website/docs/cloud-docs/api-docs/workspaces.mdx:515]: Same rewording suggestions as the table earlier.

For the Description of `data.attributes.vcs-repo.tags-regex`, I'd suggest rewording to "Terraform Cloud triggers a run when this value is present and a VCS event occurs that contains a matching Git tag for the regular expression."

For the Description of `data.attributes.working-directory`, I'd suggest rewording to "A relative path from which Terraform executes. The path defaults to your repository's root and is typically set to a subdirectory matching the environment when multiple environments exist within the same repository."

File: website/docs/cloud-docs/api-docs/workspaces.mdx [hunk 3]
```diff
@@ -119,7 +120,33 @@ _With a VCS repository_
...
+      },
+      "updated-at": "2017-11-29T19:18:09.976Z"
+    },
+    "type": "workspaces"
+  }
+}
+```
+_Using Git Tags
+
+A run will be triggered when a Git tag is pushed matching the regular expression (SemVer): `1.2.3`, `22.33.44`, etc.
```
Comment (mgarrell777) [inline website/docs/cloud-docs/api-docs/workspaces.mdx:134]: I suggest rewording to "A pushed Git tag matching the regular expression triggers a run."

**Review by mgarrell777 — APPROVED** [review]
Good stuff overall and I approve with a couple minor wording change suggestions.

Thank you once again @hashimoon for your time and work.

File: website/docs/cloud-docs/api-docs/workspaces.mdx (hunk 1 above)
Comment (judithpatudith) [reply website/docs/cloud-docs/api-docs/workspaces.mdx:89]: One tweak to the `data.attributes.working-directory` section. Maybe: 
The path defaults to your repository's root. Set it to a subdirectory matching the environment when multiple environments exist within the same repository."

**Comment by laurapacilio** [comment]
Hi @hashimoon - one thing I just thought of. This is only the API docs, but do we need to update this page as well to explain to users that this is now an option? 
https://www.terraform.io/cloud-docs/workspaces/settings/vcs#automatic-run-triggering

We may also want to update this sentence on this page:
"A workspace is linked to one branch of a VCS repository and ignores changes to other branches. You can specify which files and directories within your repository trigger runs. **Terraform Cloud can also automatically trigger runs when you create Git tags.** Refer to [Automatic Run Triggering](https://www.terraform.io/cloud-docs/workspaces/settings/vcs#automatic-run-triggering) for details.

Let me know how I can help :)

File: website/docs/cloud-docs/api-docs/workspaces.mdx (hunk 1 above)
Comment (matejrisek) [reply website/docs/cloud-docs/api-docs/workspaces.mdx:89]: Hi @mgarrell777 and @judithpatudith it's always great to get input on how to improve documentation. However working directory is not in the scope of changes this PR targets. Even thou this is a small change and it would hurt the readability of the PR, having changes unrelated to the goal of the PR makes it harder to distinguish what really matters for - in this case `tags-regex` field. Also if we'd have to revert changes we would have to manually scan the changes in order to identify the ones that are related to the `tags-regex` field and not to remove all the improvements.

WDYT

Comment (matejrisek) [reply website/docs/cloud-docs/api-docs/workspaces.mdx]: @mgarrell777 thanks for the suggestion for `tags-regex` -> applied here 0fd41d584b222d61cfc06506b7f1a00ecf8c4e13

File: website/docs/cloud-docs/api-docs/workspaces.mdx (hunk 3 above)
Comment (matejrisek) [reply website/docs/cloud-docs/api-docs/workspaces.mdx:134]: I've noticed we're changing narrative in few fields to include first-person view - e.g. `when you push`, `your repository` etc. Is this going to be a preferred path forward?

Applied @laurapacilio suggestion -> be9cee61269a3550f30371f5263f035e121841f1 -> but also left the example from before for the clarification

File: website/docs/cloud-docs/workspaces/settings/vcs.mdx [hunk 4]
```diff
@@ -65,6 +65,16 @@ You can use trigger patterns and trigger prefixes in the **Add path** field to s
...
 
+#### Trigger runs when a git tag is published
+
+This option instructs Terraform to begin new runs only for changes that have a specific tag format.
+
+The tag format can be chosen between the following options:
+- **Semantic Versioning:** It matches tags in the popular [SemVer format](https://semver.org/). (e.g. 0.4.2)
+- **Version contains a prefix:** It matches tags which have an additional prefix before the [SemVer format](https://semver.org/). (e.g. version-0.4.2)
+- **Version contains a suffix:** It matches tags which have an additional suffix after the [SemVer format](https://semver.org/). (e.g. 0.4.2-alpha)
+- **Custom Regular Expression:** You can define your own regex to be matched against tags.
```
Comment (laurapacilio) [inline website/docs/cloud-docs/workspaces/settings/vcs.mdx:76]: For SemVer, do we also accept if there is a v in front of the tag? I ask because for other things (like provider versions) I believe we recognize the tag if it's SemVer and/or we accept a v in front of the SemVer tag format - e.g.,`v0.1.2`. 

```suggestion
#### Trigger runs when a git tag is published

This option instructs Terraform to begin new runs when you publish a Git tag with a specific format.

Choose one of the following options to specify the tag format:
- **Semantic Versioning:** Tags match [SemVer format](https://semver.org/). (e.g., 0.4.2).
- **Version contains a prefix:** Tags have additional prefix before the [SemVer format](https://semver.org/). (e.g. version-0.4.2)
- **Version contains a suffix:** Tags have an additional suffix after the [SemVer format](https://semver.org/). (e.g., 0.4.2-alpha)
- **Custom Regular Expression:** Define your own regex for tag format.
```

Comment (matejrisek) [reply website/docs/cloud-docs/workspaces/settings/vcs.mdx:76]: Hey @laurapacilio I would say that for that use case we offer it through the `Version contains a prefix` option.

btw. could you please explain why do we put comma behind `e.g.` in the first and third point but not in second? And what is the grammar rule driving this?

Also isn't the period at the end of the first point redundant? If it's not we should add it to the second and third as well.

File: website/docs/cloud-docs/workspaces/settings/vcs.mdx [hunk 5]
```diff
@@ -65,6 +65,16 @@ You can use trigger patterns and trigger prefixes in the **Add path** field to s
 
 -> **Note:** Terraform Cloud triggers runs on all attached workspaces if it does not receive a list of changed files or if that list is too large to process. When this happens, Terraform Cloud may show several runs with completed plans that do not result in infrastructure changes.
 
+#### Trigger runs when a git tag is published
+
+This option instructs Terraform to begin new runs only for changes that have a specific tag format.
+
+The tag format can be chosen between the following options:
+- **Semantic Versioning:** It matches tags in the popular [SemVer format](https://semver.org/). (e.g. 0.4.2)
```
Comment (omarismail) [inline website/docs/cloud-docs/workspaces/settings/vcs.mdx:73]: Do we want to give the regex for these three values?

**Review by omarismail — APPROVED** [review]
one comment, otherwise LGTM

File: website/docs/cloud-docs/workspaces/settings/vcs.mdx (hunk 4 above)
Comment (laurapacilio) [reply website/docs/cloud-docs/workspaces/settings/vcs.mdx:76]: Hi! Thank you for asking!

## Periods
Ah thank you! This is what I get for trying to multitask while editing - I meant to move all of the commas to the end so that it's an entire sentence with the e.g. included as an example. I've updated my suggestion to fix this.

## Commas after e.g.
It's a mistake that I didn't add one after the second example - thank you for catching that!  I've updated my suggestion to fix.

In terms of explanation: I was always taught that you add a comma after e.g., which is why I made those changes here. For example - [this article](https://www.grammarly.com/blog/know-your-latin-i-e-vs-e-g/#:~:text=E.g.%20and%20i.e.%20are%20both,one%20item%20in%20your%20list.) talks about how generally you add a comma. I believe it is also recommended in many style manuals, for example, I believe the Chicago Manual of Style recommends a comma after e.g. Interestingly, as I do research, I believe there is a difference here between American style guide and British English (e.g. Oxford Manual of Style) where British English often does not include the comma.

I'm going to:
- Bring this up with technical writing so we can get something official added to the HashiCorp style guide either way
- Recommend that here we use commas, to be consistent with what we have recently added to the rest of the page.


I hope this is helpful - let me know if you have more questions. I appreciate the attention to detail here and that you didn't just blindly accept my edits!

File: website/docs/cloud-docs/workspaces/settings/vcs.mdx (hunk 5 above)
Comment (mjyocca) [reply website/docs/cloud-docs/workspaces/settings/vcs.mdx:73]: Done
